dev/Common/ReunionTelemetry.h

```diff
@@ -0,0 +1,55 @@
+﻿#pragma once
+
+#ifndef __REUNIONTELEMETRY_INCLUDED
+#define __REUNIONTELEMETRY_INCLUDED
+
+#ifdef __WIL_TRACELOGGING_H_INCLUDED
+#error "WIL Tracelogging.h must not be explicitly included when including this file"
+#endif
+
+#define _REUNION_PACKAGE_VER_RESOURCEID             10001
+#define _REUNION_EXPERIMENTATION_FLAG_RESOURCEID    10002
+
+static std::wstring s_ReunionPackageVer = L"";
```

Comment:
I vaguely wonder about having a static variable like this in header that will presumably be used in many .cpp files.
Won't this mean that each translation unit (.cpp file) gets its own copy of the variable? (ie: bloat).

Reply:
Is there an alternative?

Reply:
Yes; define the methods here in the header, and put the implementations into a .cpp file that compiles into a static lib.

Use a function local static:

```c++
inline std::wstring load_resource_string(const char* dllName, uint32_t id) {
    wil::unique_hmodule module{::LoadLibraryA(dllName)};
    if (module) {
        wchar_t buffer[256];
        LoadStringW(module, id, buffer, ARRAYSIZE(buffer));
        return buffer;
    } else {
        return { };
    }
} 

inline const wchar_t* WINAPPSDK_PACKAGE_VER()
{
    static std::wstring packageVer = load_resource_string("ProjectReunionTelemetryResource.dll", _REUNION_PACKAGE_VER_RESOURCEID);
    return packageVer.c_str();
}
```

The runtime will ensure that static initializer is called exactly once and then the result is reused to be returned to future callers.